prov/efa/src/efa_hmem.c

```diff
@@ -142,6 +133,23 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 	else
 		info->p2p_required_by_impl = true;
 
+	info->p2p_disabled_by_user = ofi_hmem_p2p_disabled();
+
+	if (info->p2p_disabled_by_user) {
+		/* Don't need to check p2p support */
+		efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_CUDA);
```

Comment:
Given this change, we should consider abstracting away the p2p check into separate function(s). With that out of the way, we can simplify the hmem info init logic:
```
info->p2p_required_by_impl = whatever;
info->p2p_disabled_by_user = IS_SYSTEM ? false : ofi_hmem_p2p_disabled();
info->p2p_supported_by_device = info->p2p_disabled_by_user ? false : is_p2p_supported(iface);
...
```